src/hooks/core/chat/useDMMessage.ts

```diff
@@ -0,0 +1,112 @@
+import {v4 as uuid} from 'uuid';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import ChannelListMemberSchema from '../../../database/schema/ChannelListMemberSchema';
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import ChatSchema from '../../../database/schema/ChatSchema';
+import UserSchema from '../../../database/schema/UserSchema';
+import {DEFAULT_PROFILE_PIC_PATH} from '../../../utils/constants';
+import useUserAuthHook from '../auth/useUserAuthHook';
+import useChatUtilsHook from './useChatUtilsHook';
+import {initChatFromPost, initChatFromPostAnon} from '../../../service/chat';
+
+type ChannelCategory = 'SIGNED' | 'ANONYMOUS';
+
+const useDMMessage = () => {
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const {signedProfileId, anonProfileId} = useUserAuthHook();
+  const {goToChatScreen} = useChatUtilsHook();
+
+  const saveChannelList = async (channel, type: ChannelCategory) => {
+    try {
+      if (channel?.members?.length === 0) return Promise.reject(Error('no members'));
+      let userTarget = channel?.better_channel_members.find(
+        (data) => data?.user?.id !== signedProfileId
+      );
+
+      if (type === 'ANONYMOUS') {
+        userTarget = channel?.better_channel_members.find(
+          (data) => data?.user?.id !== anonProfileId
+        );
+      }
+
+      const chatType = type === 'ANONYMOUS' ? 'ANON_PM' : 'PM';
+      const chatName = {
+        name: userTarget?.user?.name || userTarget?.user?.username,
+        image: userTarget?.user?.image || DEFAULT_PROFILE_PIC_PATH
+      };
+
+      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+      channel.myUserId = signedProfileId;
+      channel.targetName = chatName?.name;
+      channel.targetImage = chatName?.image;
+
+      channel.channel = {...channel};
+      const channelList = ChannelList.fromChannelAPI(channel, chatType, channel?.members);
+      await channelList.saveIfLatest(localDb);
+      refresh('channelList', 'chat', 'channelInfo', 'channelMember');
+      return goToChatScreen(channelList);
+    } catch (error) {
+      console.log('sendMessage', error);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `saveChannelList` is modifying the input parameter `channel`. This can lead to unexpected side effects if the caller of this function is not expecting `channel` to be modified. It's generally a good practice to avoid mutating function arguments directly. Instead, you could create a copy of the object and modify that.

```diff
-      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
-      channel.myUserId = signedProfileId;
-      channel.targetName = chatName?.name;
-      channel.targetImage = chatName?.image;
-      channel.channel = {...channel};
+      const updatedChannel = {...channel};
+      updatedChannel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+      updatedChannel.myUserId = signedProfileId;
+      updatedChannel.targetName = chatName?.name;
+      updatedChannel.targetImage = chatName?.image;
+      updatedChannel.channel = {...updatedChannel};
```

Also, error handling in this function could be improved. Currently, if an error occurs, it is logged to the console but not propagated further. Depending on your application's needs, you might want to throw the error after logging it, or return a specific value indicating failure.

```diff
-      console.log('sendMessage', error);
+      console.error('sendMessage', error);
+      throw error;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->